spotify_tensorflow/featran.py

```diff
@@ -57,6 +60,42 @@ def names(cls, settings_path, feature_splitter_fn=None):
         else:
             return cls.__all_names(settings)
 
+    @classmethod
+    def reorder_numpy_dataset(cls,
+                              dataset,  # type: Iterator[Dict[str, np.ndarray]]
+                              settings_path  # type: str
+                              ):
+        # type: (...) -> Iterator[OrderedDict[str, np.ndarray]]
+        """
+        Reorders a numpy dictionary so that feature keys are in the same order as those in a Featran
+        settings file.
+
+        :param dataset: A dataset created via Datasets.dict
+        :param settings_path: Path to the Featran Settings JSON Directory
+        :return: An iterator over an OrderedDict mapping feature names to Numpy arrays
+        """
+        feature_names = Featran.names(settings_path)
+        for batch in dataset:
+            yield OrderedDict((name, batch[name]) for name in feature_names)
+
+    @classmethod
+    def reorder_dataframe_dataset(cls,
+                                  dataset,  # type: Iterator[pd.DataFrame]
+                                  settings_path  # type: str
+                                  ):
+        # type: (...) -> Iterator[pd.DataFrame]
+        """
+        Reorders a pandas DataFrame so that feature columns are in the same order as those in a
+        Featran settings file.
+
+        :param dataset: A dataset created via Datasets.dataframe
+        :param settings_path: Path to the Featran Settings JSON Directory
+        :return: An iterator over new DataFrame batches with ordered columns
+        """
+        feature_names = Featran.names(settings_path)
+        for batch in dataset:
+            yield batch[feature_names]
```

Comment:
Since columns in the dataframe are named, does this operation change the structure of the underlying dataframe?

Reply:
It should just re-order the columns as expected: https://stackoverflow.com/a/29922207